### ISSUE TITLE
From the dev team: asking a favor of our community

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1722]
We've been having a number of conversations within the DeepChem development team about the future of DeepChem. We want to open out our discussion to the broader community to get your feedback on what would bring most value:

1. Do you currently use DeepChem? For what? If you used it in the past, why did you stop? Or if you considered using it at some point, why did you decide not to?
2. What are the most useful and least useful aspects of DeepChem?
3. What's the biggest pain-point of DeepChem?
4. How could DeepChem be improved to make it more relevant to your interests?
5. What's your primary deep learning framework? TensorFlow, PyTorch, or something else? If DeepChem supported an alternative framework like Pytorch Geometric (for example), would that make a difference for you?

Please leave your comments below! Alternatively, if you'd prefer to answer these questions anonymously, you can use this [google form](https://docs.google.com/forms/d/e/1FAIpQLSfFzjjBc9-T7Q1xNP_bO4fRp6wM98G3qLFgd-SjItrFCusyAA/viewform?usp=sf_link). Thank you for taking the time to help us figure out a good path forward for the project :)

**Comment by ad05bzag** [comment]
hey! 

i found out about deepchem maybe half a year ago, but only recently started experimenting more with it and with deep learning in chemometrics. So, I only have a couple of initial impressions and I won't be able to answer all the questions. 

It is great you are providing tutorials with datasets and implementing papers. It must be a lot of work to be updating everything (tutorials and code) to TF2. 

I am myself using mostly PyTorch.

I like it the library. Thank you for that.

**Comment by DinosaurInSpace** [comment]
1.  I am using DeepChem to teach myself Machine Learning/Deep Learning.  I hope to eventually use these models to advance our ability to process mass spectrometry data--my area of subject matter expertise.

2. Compared to many of the other deep-learning pipelines in the recent primary literature, DeepChem was the easiest "to get running" such that I can featurize molecular structures and compare various methods of featurization.

3. The installation on Mac did not work as described on the tutorial.  The only way I could get it to work was to create a new naked Python 2.7 environment to deal with various conflicts per this thread: https://github.com/deepchem/deepchem/issues/1396.

4. More documentation and tutorials!  I thought the Ramsundar et al "Deep Learning for the Life Sciences" book was incredibly useful.

5. Split between TensorFlow and PyTorch, based on the expertise of colleagues supporting me.

Thanks for all your hard work!

**Comment by JZuegg** [comment]
Hi,
First, you have done a wonderful work on the package... Thanks 
1. I am starting to use DeepChem again, after a break mainly due to project progress and not having had new data.
2. DeepChem is easy to start and run an 'off-the-shelf' DL model 
3a. Hyperparameter optimisation is a bit of a pain. Would be good to have it as part of the model object, using the train and validation set.. especially for Weave model for which I always have a hard time to find good parameters. 
3b. Implementing different Network structures, like a different MultiTask architectures  
4. Maybe some tools to extract and possibly visualize the hidden layer or key features (fingerprints)
5. Not an issue, yet. But I might see when I will implement a prediction service/server for people to submit SMILES/SDF's   ...

**Comment by 0ut0fcontrol** [comment]
Hi,
Thank you to DeepChem team,  @mapleleaf-soar  and I published my [first paper](https://doi.org/10.3389/fphar.2020.00069) based on ACNN model in DeepChem to show the bias in the PDBbind data set. 

We found ACNN model can achieve "state-of-the-art" in PDBbind data set using **ligand alone**.😂

I create another issue #1794 for introducing the changes we made and asking advice on how to add them into DeepChem.

1. I use DeepChem since 2017 and try DTNN and ACNN model for predicting molecular interactions.
2. Most useful: A lot of out-of-the-box models and examples.
3. Pain-point: The API change often.
4. and 5. Now I am trying pyG and DGL for graph neural network. If DeepChem supported them, does the API change again?
